cpu/avr8_common/thread_arch.c

```diff
@@ -302,21 +294,27 @@ __attribute__((always_inline)) static inline void avr8_context_save(void)
         "cli                                     \n\t"
         "push __tmp_reg__                        \n\t"
 
-#if (CHECK_RAMPZ_REG)
+#if __AVR_HAVE_RAMPZ__
         "in     __tmp_reg__, __RAMPZ__           \n\t"
         "push   __tmp_reg__                      \n\t"
 #endif
 
-#if (CHECK_RAMPDXY_REG)
+#if __AVR_HAVE_RAMPY__
         "in     __tmp_reg__, __RAMPY__           \n\t"
         "push   __tmp_reg__                      \n\t"
+#endif
+
+#if __AVR_HAVE_RAMPX__
         "in     __tmp_reg__, __RAMPX__           \n\t"
         "push   __tmp_reg__                      \n\t"
+#endif
+
+#if __AVR_HAVE_RAMPD__
```

Comment:
This is a bit more elaborate than needed, as current MCUs either have all of RAMP{D,X,Y} or none of it. But it requires no explaining and who knows, maybe it is more future proof.